### ISSUE TITLE
How to work with .env.test with Docker

### ISSUE THREAD

**Issue by ArthurZaharov** [fs/rails-base-graphql-api#141]

**Comment by AleksandrSeleznev** [comment]
It works fine for now.
Also, if I'm not mistaken, we have to take a look to `docker-compose.yml` in the `environment` section.
The issue is that Docker by default fetch `ENV` variables from `.env` during building.
And, for example, if you have some variable defined in `environment` section and this variable has two different values in `.env` and in, for example, `.env.test` files, value would have precedence which is taking place in `.env` file.

Example:

```
  environment:
    - MAILER_SENDER_ADDRESS
```
`.env` file:
`MAILER_SENDER_ADDRESS=first@example.com`

`.env.test` file:
`MAILER_SENDER_ADDRESS=second@example.com`

In the docker container, the `MAILER_SENDER_ADDRESS` variable will be set as `first@example.com`.

@EvgeniyEsaulkov could you confirm this, please?

**Comment by AleksandrSeleznev** [comment]
Here is the article about this:
https://docs.docker.com/compose/environment-variables/

**Comment by EvgeniyEsaulkov** [comment]
> @EvgeniyEsaulkov could you confirm this, please?

Absolutely. Variables defined in `docker-compose.yml` are system variables for the container and will be fetched from the custom `.env` file only using the `--env-file` option - https://docs.docker.com/compose/environment-variables/#using-the---env-file--option

Rest of variables that are loaded to Rails via `dotenv` gem are available the same way as we work without Docker.
